enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,188 @@
+
+### Non-Goals
+- The enhancement will not describe the OCM user experience.
+- The enhancement will not provide a generic way to pull information from OCM into the cluster.
+- The process will not apply to disconnected clusters.
+
+## Proposal
+
+The Insights operator already communicates with the OCM API (see [Insights Operator pulling and exposing SCA certs from the OCM API](../insights/pulling-sca-certs-from-ocm.md). This is the proposed flow for updating the `pull-secret`:
+
+1. OCM provides an API endpoint that the cluster can call to check the current global updated `pull-secret`.
+2. User starts the migration process in the OCM by providing the `pull-secret` associated with the account to which the ownership will be transferred to.
+   * The details of whether the user provides the pull secret as a means of verification, or whether a request/approval workflow is implemented, is left to the OCM/UI flow. What is pertinent here is that there is an endpoint available to a cluster entity to fetch the new `pull-secret`.
+3. OCM validates the `pull-secret` for the new account and makes it available to the cluster for some time period i.e. the new API will now return the new `pull-secret` provided by the user.
+4. The Insights operator calls this API periodically, fetches the updated `pull-secret` and compares it with the current global `pull-secret` on the cluster (only entries available in both pull-secrets are compared).
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * The updated `pull-secret` from the OCM API is an authoritative source. Insights operator replaces and adds (There should not be any new entries, but this may change in the future) the entries provided in the updated `pull-secret` from the OCM API (This means that the Insights operator cannot delete anything in the existing secret).
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots. This will require an update of the documentation in https://docs.openshift.com/container-platform/4.9/updating/update-using-custom-machine-config-pools.html#update-using-custom-machine-config-pools-about_update-using-custom-machine-config-pools.
+   We should emphasize the need of unpausing the machine config pools to succesfully apply the new `pull-secret` (credentials) to all the nodes.
+6. The Insights operator itself starts using the new `pull-secret`
+7. Cluster comes up with the new `pull-secret` and starts reporting back to OCM into the new account.
+
+Note: This process will work only on the clusters where the feature was released as part of the Insights operator and will not be backported to older versions.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+
```

Comment:
You can also link to the stories here so that it is clearly called out:
```suggestion
- [CCXDEV-6529 Insights Operator pulling updated pull-secret to transfer unclassified cluster](https://issues.redhat.com/browse/CCXDEV-6529)
- [CCX-185 Transfer of Unclassified clusters - provide pull-secret](https://issues.redhat.com/browse/CCX-185)